pkg/signup/service/signup_service.go

```diff
@@ -223,12 +234,16 @@ func (s *ServiceImpl) reactivateUserSignup(ctx *gin.Context, existing *toolchain
 // GetSignup returns Signup resource which represents the corresponding K8s UserSignup
 // and MasterUserRecord resources in the host cluster.
 // Returns nil, nil if the UserSignup resource is not found or if it's deactivated.
-func (s *ServiceImpl) GetSignup(userID string) (*signup.Signup, error) {
-
-	// Retrieve UserSignup resource from the host cluster
-	userSignup, err := s.CRTClient().V1Alpha1().UserSignups().Get(EncodeUserID(userID))
-	if err != nil {
-		if apierrors.IsNotFound(err) {
+func (s *ServiceImpl) GetSignup(userID, username string) (*signup.Signup, error) {
+	// Declare variables up front
+	var userSignup *toolchainv1alpha1.UserSignup
+	var err error
+
+	// Retrieve UserSignup resource from the host cluster, using the specified UserID and username
+	userSignup, err = s.GetUserSignup(userID, username)
```

Comment:
The same question here - why do we need to declare them up front?

Reply:
Fixed